piped/src/client.rs

```diff
@@ -7,32 +7,37 @@ pub struct PipedClient {
     pub instance: String,
 }
 
+const USER_AGENT: &'static str = "Mozilla/5.0 (Windows NT 10.0; rv:78.0) Gecko/20100101 Firefox/78.0";
+
 impl PipedClient {
-    pub fn new(httpclient: Client, instance: String) -> PipedClient {
+    pub fn new<S: AsRef<str>>(httpclient: &Client, instance: S) -> PipedClient {
         PipedClient {
-            httpclient,
-            instance,
+            httpclient: httpclient.clone(),
```

Comment:
Why clone the httpclient? :thinking:

Reply:
Because the constructor now takes a reference. As you need a object for your `PipedClient`, it will need to be cloned (converted from reference to object). Cloning a Client does not change anything with the client itself and is recommended by the authors of reqwest (as far as I understand).